frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx

```diff
@@ -1,21 +1,78 @@
+
 import React, { useState } from 'react';
+import { useParams } from 'react-router-dom';
 import Card from 'components/Card/Card';
-import TokenSelectorDeposit from 'components/TokenSelector/TokenSelector';
+import TokenSelector from 'components/TokenSelector/TokenSelector';
 import { ReactComponent as HealthIcon } from 'assets/icons/health.svg';
 import { ReactComponent as EthIcon } from 'assets/icons/ethereum.svg';
 import { validateNumberInput } from '../../../../utils/regex';
 import './AddDeposit.css';
 
 export default function AddDeposit() {
+  const { positionId } = useParams();
   const [amount, setAmount] = useState('0');
-  
+  const [selectedToken, setSelectedToken] = useState('STRK');
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+
   const handleAmountChange = (e) => {
     const value = e.target.value;
     if (validateNumberInput(value)) {
       setAmount(value);
     }
   };
 
+  const handleCancel = () => {
+    setAmount('0');
+    setSelectedToken('STRK');
+    setError(null);
+  };
+
+  const handleDeposit = async () => {
+    // if (!positionId) {
+    //   setError('Position ID is required');
+    //   return;
+    // }
+
+    try {
+      setIsLoading(true);
+      setError(null);
+
+      const response = await fetch(`/api/add-extra-deposit/${positionId}`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          position_id: positionId,
+          amount: parseFloat(amount),
+          token_symbol: selectedToken,
+        }),
+      });
+
+      if (!response.ok) {
+        const errorData = await response.json();
+        throw new Error(errorData.message || `HTTP error! status: ${response.status}`);
+      }
+
+      const data = await response.json();
+      console.log('Deposit successful:', data);
+      
+      // Reset form after successful deposit
+      setAmount('0');
+      setSelectedToken('STRK');
+      
+      // Success notification
+      alert('Deposit successful!'); 
+
+    } catch (err) {
+      setError(err.message);
```

Comment:
we don't need error state, just show error in a toast message in `onError` callback of `useMutation`